# Cephalochordate-Comparative-Genomics
Code for the genome assembly, post-processing (polishing, haplotig-purging, repeat-masking & scaffolding), transcriptome assembly, genome annotation, and multi-locus phylogenetic analyses used in the MPhil Thesis of Jaruwatana (Sodai) Lotharukpong.
Each file corresponds to the script ultimately used in each section of the MPhil Thesis:


<details>
  <summary>2.1 *De novo* genome assembly</summary>
  
    * Pre-assembly analysis
    * Hybrid assembly of *Asymmetron*
    * Hybrid assembly of *Epigonichthys*
    * Long-read assembly of *B. lanceolatum* (North Sea)
</details>

<details>
  <summary>2.2 Post-processing</summary>
  
    * Short-read polishing of *Asymmetron* & *Epigonichthys*
    * Long-read polishing of *B. lanceolatum* (North Sea)
    * Haplotig-purging of *Asymmetron*, *Epigonichthys* & *B. lanceolatum* (North Sea)
    * Repeat-masking of *Asymmetron*, *Epigonichthys* & *B. lanceolatum* (North Sea)
    * RNA-scaffolding of *Asymmetron*, *Epigonichthys* & *B. lanceolatum* (North Sea)
</details>

<details>
  <summary>2.3 Genome annotation</summary>
  
    * BRAKER annotation of *Asymmetron*, *Epigonichthys* & *Branchiostoma*
    * BRAKER re-annotation of *B. lanceolatum* (from Marlétaz et al. 2018)
    * Proteome extraction
</details>

<details>
  <summary>2.4 *De novo* transcriptome assembly</summary>
  
    * Transcriptome assembly of *B. lanceolatum* (from Banyuls-sur-Mer; RNA-seq data courtesy of Dr. Benito-Gutiérrez)
    * Transcriptome assembly of *A. lucayanum* (from Yue et al. 2014)  
</details>

<details>
  <summary>2.5 Phylogenetic reconstruction</summary>
  
    * Isoform filtering
    * Orthogroup inference
    * Visualisation of protein clusters
    * Filtering orthogroup
    * Gene tree pruning
    * Quartet-based ASTRAL analysis
    * Concatenation-based IQTREE analysis
</details>

<details>
  <summary>2.6 Evaluating gene tree discordance</summary>
  
    * Calculating quartet-based internode certainty
    * Visualising of gene tree discordance
    * Polytomy analysis
    * Analysis of the super-matrix (saturation, treeness/RCV)
    * Partition analysis
</details>
